### PR TITLE
Adding HPC recommendations

### PR DESCRIPTION
**Pull request by oZakari**
# Overview/Summary

This is re-adding the HPC Pack recommendations into the HPC Specialized workload sections which was previously removed.

## Related Issues/Work Items

Replace this with a list of related GitHub Issues and/or ADO Work Items (Internal Only)

-N/A

## This PR fixes/adds/changes/removes

1. Adding back the HPC specialized workload section
2. Adding all HPC-Pack recommendations that were previously removed from[ Azure Batch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pull/298). 

### Breaking Changes

1. None

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by oZakari** [comment]
image
image

File: docs/content/services/specialized-workloads/azure-hpc/_index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
++++
+
+The presented resiliency recommendations in this guidance include Azure High Performance Computing and associated resources and settings.
+
+## Summary of Recommendations
+
+{{< table style="table-striped" >}}
+| Recommendation | Impact | Design Area | State | ARG Query Available |
+| :------------------------------------------------ | :---------------------------------------------------------------------: | :------: | :------: | :------: |
+| [HPC-1 - Ensure File shares that stores jobs metadata are accessible from all head nodes](#hpc-1---ensure-file-shares-that-stores-jobs-metadata-are-accessible-from-all-head-nodes) | High | Resiliency/Monitoring | Preview | No |
```
Comment (ejhenry) [inline docs/content/services/specialized-workloads/azure-hpc/_index.md:17]:
```suggestion
| [HPC-1 - Ensure File shares that stores jobs metadata are accessible from all head nodes](#hpc-1---ensure-file-shares-that-stores-jobs-metadata-are-accessible-from-all-head-nodes) | High | Application Resilience | Preview | No |
```

File: docs/content/services/specialized-workloads/azure-hpc/_index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+| [HPC-2 - Automatically grow and shrink HPC Pack cluster resources](#hpc-2---automatically-grow-and-shrink-hpc-pack-cluster-resources) | Medium | System Efficiency | Preview | No |
+| [HPC-3 - Use multiple head nodes for HPC Pack](#hpc-3---use-multiple-head-nodes-for-hpc-pack) | Medium | Resiliency | Preview | No |
```
Comment (ejhenry) [inline docs/content/services/specialized-workloads/azure-hpc/_index.md:19]:
```suggestion
| [HPC-3 - Use multiple head nodes for HPC Pack](#hpc-3---use-multiple-head-nodes-for-hpc-pack) | Medium | Application Resilience | Preview | No |
```

File: docs/content/services/specialized-workloads/azure-hpc/_index.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+| [HPC-4 - Use HPC Pack Azure AD Integration or other highly available AD configuration](#hpc-4---use-hpc-pack-azure-ad-integration-or-other-highly-available-ad-configuration) | High | Resiliency | Preview | No |
```
Comment (ejhenry) [inline docs/content/services/specialized-workloads/azure-hpc/_index.md:20]:
```suggestion
| [HPC-4 - Use HPC Pack Azure AD Integration or other highly available AD configuration](#hpc-4---use-hpc-pack-azure-ad-integration-or-other-highly-available-ad-configuration) | High | Application Resilience | Preview | No |
```

File: docs/content/services/specialized-workloads/azure-hpc/_index.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+Definitions of states can be found [here]({{< ref "../../../_index.md#definitions-of-terms-used-in-aprl">}})
+
+{{< /alert >}}
+
+## Recommendations Details
+
+### HPC-1 - Ensure File shares that stores jobs metadata are accessible from all head nodes
+
+**Category: Application Resilience/Availability**
```
Comment (ejhenry) [inline docs/content/services/specialized-workloads/azure-hpc/_index.md:34]:
```suggestion
**Category: Application Resilience**
```

File: docs/content/services/specialized-workloads/azure-hpc/_index.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+{{< code lang="sql" file="code/hpc-2/hpc-2.kql" >}} {{< /code >}}
+
+{{< /collapse >}}
+
+<br><br>
+
+### HPC-3 - Use multiple head nodes for HPC Pack
+
+**Category: Resiliency**
```
Comment (ejhenry) [inline docs/content/services/specialized-workloads/azure-hpc/_index.md:93]:
```suggestion
**Category: Application Resilience**
```

File: docs/content/services/specialized-workloads/azure-hpc/_index.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+{{< code lang="sql" file="code/hpc-3/hpc-3.kql" >}} {{< /code >}}
+
+{{< /collapse >}}
+
+<br><br>
+
+### HPC-4 - Use HPC Pack Azure AD Integration or other highly available AD configuration
+
+**Category: Resiliency**
```
Comment (ejhenry) [inline docs/content/services/specialized-workloads/azure-hpc/_index.md:115]:
```suggestion
**Category: Application Resilience**
```

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Review by Rajani-Janaki-Ram — CHANGES_REQUESTED** [review]
Can we also cross link to the Batch ones (https://azure.github.io/Azure-Proactive-Resiliency-Library/services/batch/batch-accounts) in the table on this main page:

**Comment by oZakari** [comment]
@Rajani-Janaki-Ram, I have added them, please let me know if you need anything else.

**Review by Rajani-Janaki-Ram — APPROVED** [review]

**Pull request merged**